modules/net-ncc/main.tf

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  spoke_vms = flatten([
+    for spoke_key, spoke in var.spokes : [
+      for nva in spoke.nvas : {
+        ip        = nva.ip
+        vm        = nva.vm
+        vm_name   = element(split("/", nva.vm), length(split("/", nva.vm)) - 1)
+        spoke_key = spoke_key
+        spoke     = spoke
+      }
+    ]
+  ])
+}
+
+resource "google_network_connectivity_hub" "hub" {
```

Comment:
I'd leave the hub out of the module and make the ncc-ra module instead?
Can we pass it as a parameter?